### DISCUSSION TITLE
Nuxt 3 Generic type component

### DISCUSSION THREAD

**Discussion by maxdzin** [nuxt/nuxt#17272] [Questions] [answered]
Hi folks!

I wondering if there's a way to use the component by specifying the type.

What I want to reach:
FormSelectGeneric - a component where `options` prop for select input is declared as generic type:
``` js
options: {
  type: Array as PropType<T[]>,
  required: true,
},
```
and then use it in some page or another component...

There's a good [explanation and example article](https://logaretm.com/blog/generically-typed-vue-components/) on how to do that for the Vue.js project.

But I wondering if that is possible to do in Nuxt v3 (since the Nuxt importing components dynamically).

Thanks!

**Comment by Flanker72** [comment, marked as answer]
Seems you can try to drop off Auto Import feature for such components. Nuxt allows you to directly import components from external files. So, you may use your `components` dir as usual and create a separate folder (ex: `generic-components`) to store your generics. Then you may use `import GenericComponent from "~/generic-components/test.vue";` and it should work (at least in my thoughts).

**Reply by maxdzin** [reply]
@Flanker72 Exactly. Thank you!
